source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -1646,5 +1729,186 @@ namespace RendererQml
+            {
+                if (Utils::CaseInsensitiveCompare(action->GetStyle(), "positive"))
+                {
+                    bgRectangle->Property("border.color", Formatter() << buttonId << ".pressed ? '#196323' : '#1B8728'");
+                    bgRectangle->Property("color", Formatter() << buttonId << ".pressed ? '#196323' : " << buttonId << ".hovered ? '#1B8728' : 'white'");
+                    contentText->Property("color", Formatter() << buttonId << ".hovered ? '#FFFFFF' : '#1B8728'");
+                }
+                else if (Utils::CaseInsensitiveCompare(action->GetStyle(), "destructive"))
+                {
+                    bgRectangle->Property("border.color", Formatter() << buttonId << ".pressed ? '#A12C23' : '#D93829'");
+                    bgRectangle->Property("color", Formatter() << buttonId << ".pressed ? '#A12C23' : " << buttonId << ".hovered ? '#D93829' : 'white'");
+                    contentText->Property("color", Formatter() << buttonId << ".hovered ? '#FFFFFF' : '#D93829'");
+                }
+                else
+                {
+                    bgRectangle->Property("border.color", Formatter() << buttonId << ".pressed ? '#0A5E7D' : '#007EA8'");
+                    bgRectangle->Property("color", Formatter() << buttonId << ".pressed ? '#0A5E7D' : " << buttonId << ".hovered ? '#007EA8' : 'white'");
+                    contentText->Property("color", Formatter() << buttonId << ".hovered ? '#FFFFFF' : '#007EA8'");
+                }
+            }
+            else
+            {
+                bgRectangle->Property("border.color", Formatter() << buttonId << ".pressed ? '#0A5E7D' : '#007EA8'");
+                bgRectangle->Property("color", Formatter() << buttonId << ".pressed ? '#0A5E7D' : " << buttonId << ".hovered ? '#007EA8' : 'white'");
+                contentText->Property("color", Formatter() << buttonId << ".hovered ? '#FFFFFF' : '#007EA8'");
+            }
+
+            textLayout->AddChild(contentText);
+            buttonElement->Property("background", bgRectangle->ToString());
+
+            if (isShowCardButton)
+            {
+                showCardIcon = std::make_shared<QmlTag>("Image");
+                showCardIcon->Property("id", Formatter() << buttonId << "_dd");
+                showCardIcon->Property("height", Formatter() << fontSize);
+                showCardIcon->Property("width", Formatter() << fontSize);
+                showCardIcon->Property("fillMode", "Image.PreserveAspectFit");
+                showCardIcon->Property("anchors.verticalCenter", "parent.verticalCenter");
+
+                std::string file_path = __FILE__;
+                std::string dir_path = file_path.substr(0, file_path.rfind("\\"));
+                dir_path.append("\\Images\\arrow-down_12.svg");
+                std::replace(dir_path.begin(), dir_path.end(), '\\', '/');
+                showCardIcon->Property("source", "\"" + std::string("file:/") + dir_path + "\"");
+
```

Comment:
We should find a better solution to call image file. I will work on a separate PR to try get rid of this.

Reply:
We can try adding the images to qrc to make this simpler

Reply:
yes... That's the idea. Create a qrc file in the project